its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/ScannerMSBuildTest.java

```diff
@@ -996,31 +996,65 @@ void checkMultiLanguageSupportWithSdkFormat() throws Exception {
     // Begin step in MultiLanguageSupport folder
     ScannerForMSBuild scanner = TestUtils.newScanner(ORCHESTRATOR, projectDir, token)
       .addArgument("begin")
+      .setProjectDir(projectDir.toFile()) // this sets the working directory, not sonar.projectBaseDir
       .setProjectKey(folderName)
       .setProjectName(folderName)
       .setProjectVersion("1.0")
       .setProperty("sonar.sourceEncoding", "UTF-8")
+      .setProperty("sonar.verbose", "true")
       // Overriding environment variables to fallback to projectBaseDir detection
+      // This can be removed once we move to Cirrus CI.
       .setEnvironmentVariable("TF_BUILD_SOURCESDIRECTORY", "")
       .setEnvironmentVariable("TF_BUILD_BUILDDIRECTORY", "")
+      .setEnvironmentVariable("AGENT_BUILDDIRECTORY", "")
       .setEnvironmentVariable("BUILD_SOURCESDIRECTORY", "");
     ORCHESTRATOR.executeBuild(scanner);
     // Build solution inside MultiLanguageSupport/src folder
-    TestUtils.runMSBuild(ORCHESTRATOR, projectDir, "/t:Restore,Rebuild", "src/MultiLanguageSupport.sln");
+    TestUtils.runMSBuild(
+      ORCHESTRATOR,
+      projectDir,
+      // Overriding environment variables to fallback to current directory on the targets.
```

Comment:
We need to set the environment variables on every step:

## Begin step

The TFS/AZDO variables are used to calculate the `.sonarqube` directory.

## Build step

The small [targets](https://github.com/SonarSource/sonar-scanner-msbuild/blob/master/src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.ImportBefore.targets#L12) calculate where is the big target file to import it.

## End step

The TFS/AZDO variables are used to validate where the `out/` directory is.

It's a big messy, but I think it's better to be explicit about what we are doing.
It is unfortunate that we want to test non-azdo behavior (with current dir) while our ITs are running under AZDO :(